pkg/agent/datamodel/types.go

```diff
@@ -1061,6 +1091,11 @@ func (o *OrchestratorProfile) GetAPIServerEtcdAPIVersion() string {
 	return ""
 }
 
+// IsHostsConfigAgentEnabled returns true if hosts config agent is enabled
+func (o *OrchestratorProfile) IsHostsConfigAgentEnabled() bool {
+	return o.KubernetesConfig != nil && o.KubernetesConfig.PrivateCluster != nil && to.Bool(o.KubernetesConfig.PrivateCluster.EnableHostsConfigAgent)
+}
```

Comment:
We already have EnableHostsConfigAgent

Reply:
OK, I warpped the implementation of EnableHostsConfigAgent into `IsHostsConfigAgentEnabled`. revert this change to have less code change.
```
		"EnableHostsConfigAgent": func() bool {
			return cs.Properties.OrchestratorProfile.KubernetesConfig != nil &&
				cs.Properties.OrchestratorProfile.KubernetesConfig.PrivateCluster != nil &&
				to.Bool(cs.Properties.OrchestratorProfile.KubernetesConfig.PrivateCluster.EnableHostsConfigAgent)
		},
```